### PR TITLE
fix the bug about the param num_update in optimizer.py

### PR DESCRIPTION
**Pull request by 501177639**
Hi，
      this is the pulll request for my issue  https://github.com/dmlc/mxnet/issues/5059

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -143,8 +143,8 @@ def _update_count(self, index):
         """
         if index not in self._index_update_count:
             self._index_update_count[index] = self.begin_num_update
-        self._index_update_count[index] += 1
-        self.num_update = max(self._index_update_count[index], self.num_update)
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: I think this was intended for distributed training @mli

Comment (501177639) [inline python/mxnet/optimizer.py]: I think this was intended for updating the weights by kvstore? ?  But whatever,  there are still some bugs when training on a single device.

**Comment by piiswrong** [comment]
more lint errors

File: python/mxnet/optimizer.py (hunk 1 above)
Comment (mli) [inline python/mxnet/optimizer.py]: looks good to me, since `_index_update_count` is intialized with `begin_num_update`, we can drop the `max` here.

Comment (501177639) [inline python/mxnet/optimizer.py]: the problem of original code is that the first grad and others get different value of the num_update in each iteration.  Maybe you can debug it.

**Comment by piiswrong** [comment]
Can one of the admins verify this patch?

**Comment by mli** [comment]
can you resolve the conflicts? we can merge it once ci tests are passed

**Comment by mli** [comment]
I'm going to close this PR since there is no activity for more than 2 months. But please feel free to re-open it and update this branch with the master.

**Pull request closed without merging**